src/server/array_func.c

```diff
@@ -791,29 +757,28 @@ setup_arrayjob_attrs(attribute *pattr, void *pobj, int mode)
 int
 fixup_arrayindicies(attribute *pattr, void *pobj, int mode)
 {
-	int   i;
-	int   x, y, z, ct;
+	int i;
+	int x, y, z, ct;
 	char *ep;
-	job  *pjob = pobj;
+	job *pjob = pobj;
```

Comment:
Why do you need this? Why not use pobj directly?

Reply:
yes, pobj is a void pointer, to an operation like `pjob->ji_qs.ji_svrflags` we need job pointer. Also, all funcs called by this func expect job * so there also we need to typecast, so instead of doing typecase everywhere we just do in one place.